pig/src/main/java/org/jboss/pnc/bacon/pig/impl/sources/SourcesGenerator.java

```diff
@@ -77,6 +77,11 @@ public void generateSources(Map<String, PncBuild> builds, RepositoryData repo) {
             return;
         }
 
+        if (!sourcesGenerationData.getExcludeSourceBuilds().isEmpty()) {
+            log.info("Removing builds from sources generation " + sourcesGenerationData.getExcludeSourceBuilds());
+            additionalBuildsForSources.keySet().removeAll(sourcesGenerationData.getExcludeSourceBuilds());
+        }
+
```

Comment:
Looks good although I would ask if we don't want to exclude artifacts after all options of their addition is done = exclude them after all strategies to generate them are done.

Reply:
@jomrazek that's a good point, I guess it is better to exclude after all strategies are generated (as you suggested) so we are sure that what is added to the `excludeSourceBuilds` will be really removed no matter what.

I pushed a new commit with your suggestion.

Reply:
@jomrazek and the PR updating the documentation is https://github.com/project-ncl/bacon/pull/1077